frontend/views/pages/Join.vue

```diff
@@ -97,23 +97,11 @@ export default ({
   mounted () {
     // For some reason in some Cypress tests it loses the route query when initialized is called
     this.ephemeral.query = this.$route.query
-    if (syncFinished || !this.ourUsername) {
-      this.initialize()
-    } else {
-      sbp('okTurtles.events/once', LOGIN, () => this.initialize())
-    }
```

Comment:
Then you can re-arrange the code so that it looks like this:

```js
    async initialize () {
      try {
        const state = await sbp('chelonia/latestContractState', this.ephemeral.query.groupId)
        const invite = state.invites[this.ephemeral.query.secret]
        if (!invite || invite.status !== INVITE_STATUS.VALID) {
          console.error('Join.vue error: Link is not valid.')
          this.ephemeral.errorMsg = L('You should ask for a new one. Sorry about that!')
          this.pageStatus = 'INVALID'
          return
        } else if (invite.expires < Date.now()) {
          console.log('Join.vue error: Link is already expired.')
          this.ephemeral.errorMsg = L('You should ask for a new one. Sorry about that!')
          this.pageStatus = 'EXPIRED'
          return
        }
        if (this.ourUsername) {
          if (this.currentGroupId && this.$store.state.contracts[this.ephemeral.query.groupId]) {
            this.$router.push({ path: '/dashboard' })
          } else {
            await this.accept()
          }
          return
        }
```